pandas/core/generic.py

```diff
@@ -3717,6 +3717,8 @@ def to_csv(
         >>> df.to_csv('folder/subfolder/out.csv')  # doctest: +SKIP
         """
         df = self if isinstance(self, ABCDataFrame) else self.to_frame()
+        if hasattr(df, "sparse"):
```

Comment:
Is it possible to fix this later?

Reply:
Fix in what way? Open to suggestions

Reply:
Deeper in the call stack. We are formatting the values in the end before writing them to the file. Maybe there

Reply:
Moved it a few calls deeper to `CSVFormatter.__init__`